src/kOS/Suffixed/VesselTarget.cs

```diff
@@ -470,6 +473,16 @@ private void InitializeSuffixes()
             return crew;
         }
 
+        public MessageQueueStructure GetMessages()
+        {
+            if (Shared.Vessel.id != Vessel.id)
+            {
+                throw new KOSException("You can only access the message queue of the current vessel");
```

Comment:
I'd much rather have it implemented as a separate exception now while we're still looking at the code, rather than merging it and then forgetting it later on.  If and when we ever do more with exceptions in kerboscript, we may have a big task on our hands to go back through piles and piles of code to turn all the generic ``KOSException``s into specific exceptions.  It just makes more sense to have them already split out into separate types when they first get written, for all new code.  This is because at the time you write the code it's more fresh in your mind what the exception is about than months later when refactoring things.